extensions/interactions/CodeRepl/directives/CodeRepl.js

```diff
@@ -343,6 +343,10 @@ oppia.factory('codeReplRulesService', [
         var normalizedError = $filter('normalizeWhitespace')(answer.error);
         var normalizedSnippet = $filter('normalizeWhitespace')(inputs.x);
         return normalizedError.indexOf(normalizedSnippet) !== -1;
+      },
+      TrainingDataClassification: function(answer, inputs) {
```

Comment:
I don't think it makes sense to add a new rule here (or change this file at all). This service is meant to be for hard rules only, right?

Instead, the general code handling answer groups and outcomes should contain all the logic for "if interaction supports training data, do X, then do Y". There shouldn't be a need to modify these hard rule sections at all.

Reply:
We need this equality check somewhere as answer object type of each interaction is different.

> Instead, the general code handling answer groups and outcomes should contain all the logic for "if interaction supports training data, do X, then do Y". There shouldn't be a need to modify these hard rule sections at all.

I don't understand this. What you mean by general code handling answer groups and outcomes?

Reply:
@seanlip I have explained here why a general method for equality checking would not work in my comment above.

Reply:
Done